### PR TITLE
DropContainer height calculation updated to handle edge cases

### PR DESCRIPTION
**Pull request by mrebb**
DropContainer height calculation updated to handle edge cases

#### What does this PR do?
Handle edge cases to flip the select or menu control when it is within certain threshold. 

#### Where should the reviewer start?
DropContainer.js

#### What testing has been done on this PR?
Storybook

#### How should this be manually tested?
Storybook

#### What are the relevant issues?
#2994 #3008 

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
Yes

#### Is this change backwards compatible or is it a breaking change?
Yes

### CODE REVIEW

File: src/js/components/Drop/DropContainer.js [hunk 1]
```diff
@@ -155,7 +155,12 @@ class DropContainer extends Component {
           top = targetRect.bottom;
         }
 
-        if (windowHeight === top || windowHeight === targetRect.bottom) {
+        // Calculate visible area underneath the control w.r.t window height
```
Comment (ericsoderberghp) [inline src/js/components/Drop/DropContainer.js]: I'm trying to understand how the comment and the code go together. If the targetRect.bottom is 50 and the window height is 200, then the visibleAreaThreshold is 75. I'm trying to understand how 75 is either "the visible area underneath the control" or a "threshold". In the code further below 20 seems like a "threshold". Can you clarify a bit?

Comment (mrebb) [reply src/js/components/Drop/DropContainer.js]: If visibleAreaThreshold is 75% then it doesn't qualify for flipping the control. Consider that targetRect.bottom is 400  and window height is 480 then visibleAreaThreshold is close to 17% which means that it is within threshold limit that is defined here and there is not enough room to show the drop container, hence flip the control.

Comment (ericsoderberghp) [reply src/js/components/Drop/DropContainer.js]: Ah, so the 17% is a measure of the percentage of the available window height below the drop target. Can we call it something like `percentVisibleAreaBelow`?

Also, we should add a comment indicating why `20` was chosen as the threshold percentage.

Comment (mrebb) [reply src/js/components/Drop/DropContainer.js]: Updated as per your comments. Thanks for your review!!

**Pull request merged**